switchcraft/data/computercraft/lua/rom/programs/http/pastebin.lua

```diff
@@ -155,15 +165,37 @@ if sCommand == "put" then
   writeCol(colors.lightGray, "Connecting to ")
   writeCol(colors.lime, SC_PASTE_NAME)
   writeCol(colors.lightGray, "... ")
-
-  local response, err, errResponse = http.post(
+  if not tArgs[3] then
+   response, err, errResponse = http.post(
     SC_PASTE_ENDPOINT .. "/api/v1/pastes" ..
     "?language=lua" ..
     "&name=" .. textutils.urlEncode(sName),
     sText,
     { ["Content-Type"] = "text/plain" }
-  )
-
+    )
+  else
+  if tArgs[4] == nil then
+    vis = "PUBLIC"
+  else
+    vis = tArgs[4]
+  end
```

Comment:
Currently, this program does not support API tokens. All pastes are anonymous. SCPaste does not, and will never have, public anonymous paste listings. Therefore, anonymous pastes can only ever be unlisted. This option does nothing and should be removed.